### ISSUE TITLE
Compile error for double colon

### ISSUE THREAD

**Issue by ktsakas** [jcaesar/structstruck#6]
I was writing a proc macro using the `syn` library and had the following code:

```
structstruck::strike! {
    struct ItemDefine {
        semantic_token: keywords::semantic,
                                ^---- errors here
        ident: Ident,
        semantic_fields: struct {
            brace_token: token::Brace,
                              ^---- errors here
            fields: Vec<SemanticField>,
        }
    }
}
```

In the output I see:

> structstruck error: Colon in top level of type expression. Did you forget a comma somewhere?

This should be valid I was able to do this in my structs before nesting them.

PS. I am getting around this by using `type Brace = token::Brace`, but it makes things harder to read.

**Comment by jcaesar** [comment]
Uff, painful oversight on my part when trying to fix #4. Can you verify that your code works with structstruck 0.4.1?

**Comment by ktsakas** [comment]
I was playing around in my local machine so I don't have the code anymore, but from [the test case in the commit ](https://github.com/jcaesar/structstruck/commit/b37160478807be6f2f8a240d6483ee717b365669) the fix looks good.
